src/main/scala/com/github/simplesteph/ksm/parser/YamlAclParser.scala

```diff
@@ -0,0 +1,156 @@
+                          clusters: Option[Map[String, List[String]]])
+
+  case class AclYaml(users: Map[String, AclResources])
+
+  def getResource(resourceName: String, resourceType: ResourceType): Resource = {
+
+    if (resourceName.trim.endsWith("*") && resourceName.trim != "*")
+      new Resource(resourceType, resourceName.trim.dropRight(1), PatternType.PREFIXED)
+    else new Resource(resourceType, resourceName.trim, PatternType.LITERAL)
+  }
+
+  def parseResource(principal: KafkaPrincipal,
+                    resourceType: ResourceType,
+                    resources: Map[String, List[String]]
+                   ): SourceAclResult = {
+
+    val parsed = resources.toList.flatMap({
+      case (resourceName, resourceAcls) => {
+        val resource = getResource(resourceName, resourceType)
+        resourceAcls.map(acl => parseAcl(principal, resource, acl))
+      }
+    })
+
+    val acls = parsed.filter(_.isValid).flatMap(_.getOrElse(List())).toSet
+    val errs = parsed.filter(_.isInvalid).map(_.toEither.left.get).map(err => Try(err))
+    SourceAclResult(acls, errs)
+  }
+
+  val clusterWildcard = Resource(Cluster, "*", PatternType.LITERAL)
+
+  def parseAcl(principal: KafkaPrincipal, resource: Resource, acl: String): Validated[YamlParserException, List[(Resource, Acl)]] = {
+    acl.split(",") match {
+      case Array(operation, permissionType, host) =>
+        Try {
+          List(
+            (resource, Acl(principal, PermissionType.fromString(permissionType), host, Operation.fromString(operation)))
+          )
+        }
+          .toEither
+          .leftMap(error => new YamlParserException(s"Could not parse ACL '$acl' for principal '$principal' and resource '$resource'", error))
+          .toValidated
+      case Array(abbrev) => (resource.resourceType, abbrev.toLowerCase()) match {
+        case (Topic, "admin") =>
+          Valid(
+            List(
+              (resource, Acl(principal, Allow, "*", Describe)),
+              (resource, Acl(principal, Allow, "*", Create)),
+              (resource, Acl(principal, Allow, "*", Delete)),
+              (resource, Acl(principal, Allow, "*", Write)),
+              (resource, Acl(principal, Allow, "*", Read)),
+              (clusterWildcard, Acl(principal, Allow, "*", Create))
+            )
+          ).toEither
+            .toValidated
+        case (Topic, "consume") =>
+          Valid(
+            List(
+              (resource, Acl(principal, Allow, "*", Describe)),
+              (resource, Acl(principal, Allow, "*", Read))
+            )
+          )
+            .toEither
+            .toValidated
+        case (Topic, "produce") =>
+          Valid(
+            List(
+              (resource, Acl(principal, Allow, "*", Describe)),
+              (resource, Acl(principal, Allow, "*", Create)),
+              (resource, Acl(principal, Allow, "*", Write)),
+              (clusterWildcard, Acl(principal, Allow, "*", Create))
```

Comment:
I don't think that is needed. Do you have a reference?

Reply:
If you mean the cluster permission is for the topic creation, we can challenge the concept